core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -130,31 +130,33 @@ inline void check_shmem_request(CudaInternal const* cuda_instance, int shmem) {
 // KernelFuncPtr does not necessarily contain that type information.
 template <class DriverType, class LaunchBounds, class KernelFuncPtr>
 const cudaFuncAttributes& get_cuda_kernel_func_attributes(
-    int cuda_device, const KernelFuncPtr& func) {
+    const CudaInternal* cuda_instance, const KernelFuncPtr& func) {
```

Comment:
Why a ptr instead of `const CudaInternal&` taking a const& shouldn't increment ref count either or?

Reply:
I did this because that is how the other functions in `Kokkos_Cuda_KernelLaunch.hpp` pass the instance. I'd prefer `const&`, I could change them all.